.github/workflows/test-build.yaml

```diff
@@ -0,0 +1,32 @@
+name: Run tests
+
+on: pull_request
+
+jobs:
+  run_test:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Build test image
+        uses: docker/build-push-action@v2
+        with:
+          context: .
+          file: infra/Dockerfile
+          push: false
+          target: "test"
```

Comment:
In this build step prior we chose the `target: "test"` for the tag we attempt in the next step.